Add installation requirements docs (#448)
rancher/rke2: docs/install/requirements.md

RKE2 is very lightweight, but has some minimum requirements as outlined below.

## Prerequisites

Two nodes cannot have the same hostname.

> If all your nodes have the same hostname, use `--node-name` or `$RKE2_NODE_NAME` for each node you add to the cluster to have a different node name.

Reviewer: i'm trying to make reference to config file params rather than cmd line args or env vars
```suggestion
If all your nodes have the same hostname, set the `node-name` parameter in the RKE2 config file for each node you add to the cluster to have a different node name.
```

Author: got it, will fix

## Operating Systems

RKE2 is officially supported and tested on the following operating systems and their subsequent non-major releases:

*    Ubuntu 18.04 (amd64)
*    Ubuntu 20.04 (amd64)
*    CentOS/RHEL 7.8 (amd64)
*    CentOS/RHEL 8.x (amd64)

## Hardware

Hardware requirements scale based on the size of your deployments. Minimum recommendations are outlined here.

*    RAM: 512MB Minimum (we recommend at least 1GB)
*    CPU: 1 Minimum

#### Disks

> RKE2 performance depends on the performance of the database, and since RKE2 runs etcd embeddedly and it stores the data dir on disk. we recommend using an SSD when possible to ensure optimal performance.

Reviewer: ```suggestion
RKE2 performance depends on the performance of the database, and since RKE2 runs etcd embeddedly and it stores the data dir on disk, we recommend using an SSD when possible to ensure optimal performance.
```

## Networking

> The RKE2 server needs port 6443 and 9345 to be accessible by the nodes.

Reviewer: ```suggestion
The RKE2 server needs port 6443 and 9345 to be accessible by other nodes in the cluster.
```

> The RKE2 server needs port 6443 and 9345 to be accessible by the nodes.

Reviewer 2: Should we include what uses each port?

Author: I elaborated on that in the table below

> The nodes need to be able to reach other nodes over UDP port 8472 when Flannel VXLAN is used.

Reviewer: ```suggestion
All nodes need to be able to reach other nodes over UDP port 8472 when Flannel VXLAN is used.
```

If you wish to utilize the metrics server, you will need to open port 10250 on each node.

> > **Important:** The VXLAN port on nodes should not be exposed to the world as it opens up your cluster network to be accessed by anyone. Run your nodes behind a firewall/security group that disables access to port 8472.

Reviewer: I dont like how the sections that start with `>` are rendered on our docs site.

```suggestion
**Important:** The VXLAN port on nodes should not be exposed to the world as it opens up your cluster network to be accessed by anyone. Run your nodes behind a firewall/security group that disables access to port 8472.
```
